### ISSUE TITLE
Feature request: nudge recorders to pick a Species over a Genus

### ISSUE THREAD

**Issue by DavidHepper** [BiologicalRecordsCentre/iRecord#1689]
We are still finding too many recorders picking a genus when there's only one UK species in that genus. This is typically because the 'species' name they are entering shows the genus as the first pick-list option. e.g.: 
![image](https://github.com/BiologicalRecordsCentre/iRecord/assets/24915271/ba9b2d41-8401-4d2b-9cd4-a4dfc5ce2374)

I suggest adding a warning triangle to all entries of taxon level higher than Species. There is plenty of room to the right of the names, as shown above. This requires no changes to the picklist code and no updates to existing records.

The behaviour here should be matched in the App.

You might consider adding a similar warning to subspecies, hybrids, etc. - see #669, #837 - but these are not relevant to Odonata.

**Comment by sacrevert** [comment]
We have this problem in the NPMS, and @andrewvanbreda and @kazlauskis have plans to push genera below species options in the taxon picker drop downs. It would be useful if this type of thing was implemented generally in the iform module. I agree that infraspecific taxa and hybrids should also be treated like this. A workaround we are using in the NPMS app is to include a column of record frequency in the warehouse as an index of commonness, and then using this to order options within genera (so, autocomplete options are listed by order of encounter probability)

**Comment by DavidHepper** [comment]
Pushing other taxa below Species in drop down pick-list list might be useful but could be more complicated and not be appropriate if there are groups where Genus-level ID is preferred. 
My suggestion would be much simpler to implement and less likely to have side-effects. Rather than a warning triangle a colour-coded symbol could indicate the taxonomic level of each pick-list entry, e.g. 'SPECIES' in a green lozenge and 'GENUS' in a yellow one, right-aligned.

**Comment by sacrevert** [comment]
Some taxon groups (difficult vascular plants) already have warning triangles, so this would complicate visual presentation in those cases. People could still scroll for genus if required. For vascular plants most cases where genus would be suitable also species aggregate options (e.g. Taraxacum). Probably requires wider discussion across all schemes anyway

**Comment by mikeprince** [comment]
A simple way to distinguish different taxa levels would be to use indentation. I don't know the reasons for having scientific names on a separate line: this also makes it look more cluttered as well as showing less options to the user at one time. Both of these would be very simple improvements in my opinion.
![image](https://github.com/BiologicalRecordsCentre/iRecord/assets/158171/882d97b8-5b54-408f-8203-05f2aec932eb)
This probably only makes sense with genus being shown before species. Understandably some groups might prefer genus below species, but this is likely to confuse if it isn't done consistently

**Comment by DavidHepper** [comment]
Yes, that kind of inconsistency is what I was trying to avoid. Also, IMO indentation would be a bit too subtle.

**Comment by johnvanbreda** [comment]
Rather than change the sort order, which I think will always have both positives and negatives for certain use cases, I think @DavidHepper's suggestion of a visual indicator helping you spot when you've selected a rank other than species is a good one and it should not cause too much clutter if carefully designed. 

It's the taxon group that goes on the next line, we could reformat this onto the same line quite easily if that's the consensus (maybe using square brackets or something like that).

**Comment by sacrevert** [comment]
@johnvanbreda If we implement tags like "genus" and "species", will we do this for all UKSI ranks, or just these? Could future development of the iform have optional "frequency" ordering that could be implemented for projects/forms that wanted it? Seems like it would be straightforward to have a table of taxon frequency in the warehouse that was updated every year or so and which could be selected by people setting up forms if desired. Likewise, perhaps there could be an iform option to promote species (rather than demoting some set of other taxonomic levels) available to projects who want it.

Also, I don't think indentation is likely to be the solution, as we have things like Heracleum sphondylium ssp. sphondylium var. sphondylium f. sphondylium in the UKSI, which will presumably lead to display issues in the app, if not also on the website. Also, if anything, I think it is likely to encourage accidental recording at the genus level, rather than the reverse, for those cases where it currently occurs.

**Comment by mikeprince-ne** [comment]
It would be good to mockup some ideas of what we could do, including where there are very long names and how that would work on narrower displays. The use of symbols or extra tags could easily add more confusion rather than simplify things if we're not careful.
Sorting by frequency has pros and cons, one of the latter being that users get used to it for common entries and hence surprised/confused when the frequency changes and hence the order does, potentially leading to incorrect selections.
@johnvanbreda Scientific name and taxon group are both put on new lines currently, as per David's first screenshot.

**Comment by sacrevert** [comment]
@johnvanbreda it's worth noting that the dropdown appears to interact with the "What you record" settings in one's account profile in ways I don't understand. When I had nothing set the first autocomplete option in the casual record form on iRecord was "Knapweed [Centaurea]" at the genus level; after selecting all taxon groups related to vascular plant and bryophyte recording in my account profile the genus level has disappeared from my dropdown (although it is a member of "flowering plant" in the UKSI) and my first option is now the "Knapweed Bell moth [Epiblema cirsiana]" (!) 

@mikeprince-ne I was suggesting that the various options would be available to all projects, and that they would be checkbox options to enable or disable within the iform module. As I understand it, the iform is used far more widely than just iRecord, so variable presentation across projects is standard (e.g. NPMS has its own use of the forms as compared to iRecord). If resource is going to be spent on coding up mockups (impacting websites and apps), then it might be just as much resource to just implement them as available options, then the decision here is just which to use for iRecord instance.

**Comment by johnvanbreda** [comment]
@sacrevert please can you send me the exact list of groups selected, which form you were using and what the exact search string was when it failed to find Knapweed at the Genus level as it didn't do this when I tested just now.

@mikeprince-ne you are correct, thanks. It's a simple configuration to change the formatting once we get consensus.

@kitenetter can you give me some indication of priorities if you want me to mock up some options?

**Comment by sacrevert** [comment]
@johnvanbreda Maybe it was something else, as I can't reproduce. I'm surprised that "What you record" has no effect on dropdowns though -- what is that info used for?

I agree that having the scientific and common names on the same row would be helpful. I don't know whether the change being suggested here is just for iRecord or as default behaviour for all websites using indicia iform -- would be useful (at least for me) to clarify.

**Comment by johnvanbreda** [comment]
@sacrevert What you record should indeed alter the filter on the search. I'd suggest any change to iRecord would become the default for Indicia.

**Comment by sacrevert** [comment]
@johnvanbreda "What you record" doesn't seem to change anything in terms of the dropdown on the casual form for me -- shall I create a separate issue? Or are do these changes take a while to have effect (I hadn't edited them before yesterday)
